ui-service/src/main/frontend/hotel-booking-ui/src/components/search-components/HotelSearchSectionComponent.tsx

```diff
@@ -1,257 +1,280 @@
-        const {name, value, type, checked} = e.target;
-        setFormData((prevState) => ({
-            ...prevState,
-            [name]: type === "checkbox" ? checked : value,
-        }));
-    };
+    window.addEventListener("scroll", handleScroll);
 
-    const handleDestinationChange = (
-        index: number,
-        e: React.ChangeEvent<HTMLInputElement>
-    ) => {
-        setFormData(prevState => ({
-            ...prevState,
-            destinationAndDates: prevState.destinationAndDates.map((item, i) =>
-                i === index ? {...item, destination: e.target.value} : item
-            ),
-        }));
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
     };
+  }, []);
 
+  const handleDateChange = (
+    index: number,
+    dates: [Date | null, Date | null]
+  ) => {
+    const [start, end] = dates;
+    setFormData((prevState) => ({
+      ...prevState,
+      destinationAndDates: prevState.destinationAndDates.map((item, i) =>
+        i === index
+          ? {
+              ...item,
+              checkInDate: start || undefined,
+              checkOutDate: end || undefined,
+            }
+          : item
+      ),
+    }));
+  };
 
-    const handleHotelTypeChange = (type: "single" | "multi") => {
-        if (type === "multi") {
-            setFormData((prevState) => ({
-                ...prevState,
-                destinationAndDates: [
-                    {destination: "", startDate: today, endDate: tomorrow},
-                    {
-                        destination: "",
-                        startDate: undefined,
-                        endDate: undefined,
-                    },
-                ],
-                hotelType: type,
-            }));
-        } else {
-            setFormData((prevState) => ({
-                ...prevState,
-                destinationAndDates: [
-                    {destination: "", startDate: today, endDate: tomorrow},
-                ],
-                hotelType: type,
-            }));
-        }
-    };
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value, type, checked } = e.target;
+    setFormData((prevState) => ({
+      ...prevState,
+      [name]: type === "checkbox" ? checked : value,
+    }));
+  };
 
-    const handleChildAgeChange = (childId: number, age: number | null) => {
-        setFormData(prevState => ({
-            ...prevState,
-            childrenAges: prevState.childrenAges.map(child =>
-                child.id === childId ? { ...child, age } : child
-            )
-        }));
-    };
+  const handleDestinationChange = (
+    index: number,
+    e: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setFormData((prevState) => ({
+      ...prevState,
+      destinationAndDates: prevState.destinationAndDates.map((item, i) =>
+        i === index
+          ? {
+              ...item,
+              location: {
+                city: e.target.value,
```

Comment:
This might not be true. The entered value needs to be parsed, and not entirely the city